controllers/clusterurlmonitor/clusterurlmonitor_supplement.go

```diff
@@ -118,6 +120,8 @@ func (s *ClusterUrlMonitorReconciler) EnsureMonitorAndDependenciesAbsent(cluster
 
 func (s *ClusterUrlMonitorReconciler) EnsureFinalizerSet(clusterUrlMonitor v1alpha1.ClusterUrlMonitor) (utilreconcile.Result, error) {
 	if s.Common.SetFinalizer(&clusterUrlMonitor, FinalizerKey) {
+		// ignore the output as we want to remove the PrevFinalizerKey anyways
+		s.Common.DeleteFinalizer(&clusterUrlMonitor, PrevFinalizerKey)
```

Comment:
What if we fail to delete the prevfinalizer?

Reply:
the `DeleteFinalizer` is a local command only, so it just removes from an array if the item exists or keeps it empty

also the `s.Common.UpdateMonitorResource` returns the error if something goes sideways